tracing-subscriber/src/reload.rs

```diff
@@ -36,7 +36,34 @@ pub struct Layer<L, S> {
     _s: PhantomData<fn(S)>,
 }
 
-/// Allows reloading the state of an associated `Layer`.
+/// Allows reloading the state of an associated [`Layer`](crate::layer::Layer).
+///
+/// # Examples
+///
+/// Reloading a [`Filtered`](crate::filter::Filtered) layer to change the filter at runtime.
+///
+/// ```
+/// # use tracing::info;
+/// # use tracing_subscriber::{filter,fmt,reload,Registry,prelude::*};
+/// # fn main() {
+/// let (filtered_layer, reload_handle) =
+///     reload::Layer::new(fmt::Layer::default().with_filter(filter::LevelFilter::WARN));
+/// #
+/// # // specifying the Registry type is required
+/// # let _: &reload::Handle<filter::Filtered<fmt::Layer<Registry>,
+/// # filter::LevelFilter, Registry>,Registry>
+/// # = &reload_handle;
+/// #
+/// info!("This will be ignored");
+/// reload_handle.modify(|layer| *layer.filter_mut() = filter::LevelFilter::INFO);
+/// info!("This will be logged");
+/// # }
+/// ```
```

Comment:
it would be nice to also have an example of reloading a global filtering `Layer`...i'm surprised we don't have that already. but we should probably add that in a follow-up PR, rather than here.